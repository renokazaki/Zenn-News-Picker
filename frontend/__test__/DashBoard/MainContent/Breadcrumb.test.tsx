import { render, screen } from "@testing-library/react";
import React from "react";
import BreadcrumbComponents from "../../../src/components/dashboard/MainContent/Header/Breadcrumb";
import "@testing-library/jest-dom";

describe("Breadcrumb", () => {
  let container: HTMLElement;

  test("日付がある場合、正しく表示されること", () => {
    const testDate = new Date("2025-09-15");
    const renderResult = render(<BreadcrumbComponents date={testDate} />);
    container = renderResult.container;

    expect(container).toBeDefined();
    expect(screen.getByText(testDate.toLocaleDateString())).toBeInTheDocument();
  });

  test("日付がない場合、デフォルトテキストが表示されること", () => {
    const renderResult = render(<BreadcrumbComponents date={undefined} />);
    container = renderResult.container;

    expect(container).toBeDefined();
    expect(screen.getByText("All Day's News")).toBeInTheDocument();
  });

  test("スナップショットと一致すること", () => {
    const testDate = new Date("2025-09-15");
    const { container } = render(<BreadcrumbComponents date={testDate} />);
    expect(container).toMatchSnapshot();
  });
});
