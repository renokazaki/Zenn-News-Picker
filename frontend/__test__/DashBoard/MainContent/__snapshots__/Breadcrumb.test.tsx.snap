// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Breadcrumb スナップショットと一致すること 1`] = `
<div>
  <nav
    aria-label="breadcrumb"
    data-slot="breadcrumb"
  >
    <ol
      class="text-muted-foreground flex flex-wrap items-center gap-1.5 text-sm break-words sm:gap-2.5"
      data-slot="breadcrumb-list"
    >
      <li
        class="inline-flex items-center gap-1.5"
        data-slot="breadcrumb-item"
      >
        <span
          aria-current="page"
          aria-disabled="true"
          class="text-foreground text-sm font-medium"
          data-slot="breadcrumb-page"
          role="link"
        >
          2025/9/15
        </span>
      </li>
    </ol>
  </nav>
</div>
`;
